tests/integration_tests/test_lightgbm.py

```diff
@@ -37,18 +48,27 @@ def test_lightgbm_pruning_callback():
     study = optuna.create_study(pruner=DeterministicPruner(True))
     study.optimize(objective, n_trials=1)
     assert study.trials[0].state == optuna.structs.TrialState.PRUNED
+    study.optimize(partial(objective, cv=True), n_trials=1)
+    assert study.trials[0].state == optuna.structs.TrialState.PRUNED
```

Comment:
An instance of `Study` accumulates each optimization results.
So at this point, `study` has two trial results and you should assert against the second trial (i.e., `study.trials[1]`).

I think you can write your tests more briefly by using `@pytest.mark.parametrize` as follows:
```diff
diff --git a/tests/integration_tests/test_lightgbm.py b/tests/integration_tests/test_lightgbm.py
index a0b1cd3..b650d72 100644
--- a/tests/integration_tests/test_lightgbm.py
+++ b/tests/integration_tests/test_lightgbm.py
@@ -7,6 +7,10 @@ from optuna.integration.lightgbm import LightGBMPruningCallback
 from optuna.testing.integration import DeterministicPruner
 
 
+# If `True`, `lgb.cv(..)` will be used in the test, otherwise `lgb.train(..)` will be used.
+CV_FLAGS = [True]
+
+
 def test_lightgbm_pruning_callback_call():
     # type: () -> None
 
@@ -42,31 +46,23 @@ def test_lightgbm_pruning_callback_call():
         pruning_callback(env_cv)
 
 
-def test_lightgbm_pruning_callback():
-    # type: () -> None
+@pytest.mark.parametrize('cv', CV_FLAGS)
+def test_lightgbm_pruning_callback(cv):
+    # type: (bool) -> None
 
     study = optuna.create_study(pruner=DeterministicPruner(True))
-    study.optimize(objective, n_trials=1)
-    assert study.trials[0].state == optuna.structs.TrialState.PRUNED
-    study.optimize(partial(objective, cv=True), n_trials=1)
+    study.optimize(partial(objective, cv=cv), n_trials=1)
     assert study.trials[0].state == optuna.structs.TrialState.PRUNED
 
     study = optuna.create_study(pruner=DeterministicPruner(False))
-    study.optimize(objective, n_trials=1)
-    assert study.trials[0].state == optuna.structs.TrialState.COMPLETE
-    assert study.trials[0].value == 1.
-    study.optimize(partial(objective, cv=True), n_trials=1)
+    study.optimize(partial(objective, cv=cv), n_trials=1)
     assert study.trials[0].state == optuna.structs.TrialState.COMPLETE
     assert study.trials[0].value == 1.
 
     # Use non default validation name.
     custom_valid_name = 'my_validation'
     study = optuna.create_study(pruner=DeterministicPruner(False))
-    study.optimize(lambda trial: objective(trial, valid_name=custom_valid_name), n_trials=1)
-    assert study.trials[0].state == optuna.structs.TrialState.COMPLETE
-    assert study.trials[0].value == 1.
-    study.optimize(lambda trial: objective(trial, valid_name=custom_valid_name, cv=True),
-                   n_trials=1)
+    study.optimize(lambda trial: objective(trial, valid_name=custom_valid_name, cv=cv), n_trials=1)
     assert study.trials[0].state == optuna.structs.TrialState.COMPLETE
     assert study.trials[0].value == 1.
```